src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -0,0 +1,401 @@
+                member x.Ingest(ms) =
+                    System.Threading.Interlocked.Increment(&x.count) |> ignore
+                    System.Threading.Interlocked.Add(&x.ms, ms) |> ignore
+            type LogSink() =
+                static let epoch = System.Diagnostics.Stopwatch.StartNew()
+                static member val Read = Counter.Create() with get, set
+                static member val Write = Counter.Create() with get, set
+                static member val Resync = Counter.Create() with get, set
+                static member Restart() =
+                    LogSink.Read <- Counter.Create()
+                    LogSink.Write <- Counter.Create()
+                    LogSink.Resync <- Counter.Create()
+                    let span = epoch.Elapsed
+                    epoch.Restart()
+                    span
+                interface Serilog.Core.ILogEventSink with
+                    member _.Emit logEvent = logEvent |> function
+                        | MetricEvent (Read stats) -> LogSink.Read.Ingest stats
+                        | MetricEvent (Write stats) -> LogSink.Write.Ingest stats
+                        | MetricEvent (Resync stats) -> LogSink.Resync.Ingest stats
+                        | MetricEvent (Rollup _) -> ()
+                        | _ -> ()
+
+        /// Relies on feeding of metrics from Log through to Stats.LogSink
+        /// Use Stats.LogSink.Restart() to reset the start point (and stats) where relevant
+        let dump (log : ILogger) =
+            let stats =
+              [ "Read", Stats.LogSink.Read
+                "Write", Stats.LogSink.Write
+                "Resync", Stats.LogSink.Resync ]
+            let logActivity name count lat =
+                log.Information("{name}: {count:n0} requests; Average latency: {lat:n0}ms",
+                    name, count, (if count = 0L then Double.NaN else float lat/float count))
+            let mutable rows, totalCount, totalMs = 0, 0L, 0L
+            for name, stat in stats do
+                if stat.count <> 0L then
+                    totalCount <- totalCount + stat.count
+                    totalMs <- totalMs + stat.ms
+                    logActivity name stat.count stat.ms
+                    rows <- rows + 1
+            // Yes, there's a minor race here between the use of the values and the reset
+            let duration = Stats.LogSink.Restart()
+            if rows > 1 then logActivity "TOTAL" totalCount totalMs
+            let measures : (string * (TimeSpan -> float)) list = [ "s", fun x -> x.TotalSeconds(*; "m", fun x -> x.TotalMinutes; "h", fun x -> x.TotalHours*) ]
+            let logPeriodicRate name count = log.Information("rp{name} {count:n0}", name, count)
+            for uom, f in measures do let d = f duration in if d <> 0. then logPeriodicRate uom (float totalCount/d |> int64)
+
+
+[<RequireQualifiedAccess; NoEquality; NoComparison>]
+type EsSyncResult = Written of int64 | ConflictUnknown
```

Comment:
MdbSyncResult (but push down per comments in other file re removing WrongExpectedVersion)